internal/service/federatedsettingsorgrolemapping/model_federated_settings_org_role_mapping.go

```diff
@@ -37,12 +39,71 @@ func FlattenRoleAssignments(roleAssignments []*matlas.RoleAssignments) []map[str
 
 		for i := range roleAssignments {
 			roleAssignmentsMap[i] = map[string]any{
-				"group_id": roleAssignments[i].GroupID,
-				"org_id":   roleAssignments[i].OrgID,
-				"role":     roleAssignments[i].Role,
+				"group_id": roleAssignments[i].GetGroupId(),
+				"org_id":   roleAssignments[i].GetOrgId(),
+				"role":     roleAssignments[i].GetRole(),
 			}
 		}
 	}
 
 	return roleAssignmentsMap
 }
+
+func expandRoleAssignments(d *schema.ResourceData) *[]admin.RoleAssignment {
+	var roleAssignments []admin.RoleAssignment
+
+	if v, ok := d.GetOk("role_assignments"); ok {
+		if rs := v.(*schema.Set); rs.Len() > 0 {
+			for _, r := range rs.List() {
+				roleMap := r.(map[string]any)
+
+				for _, role := range roleMap["roles"].(*schema.Set).List() {
+					roleAssignment := admin.RoleAssignment{
+						OrgId:   conversion.StringPtr(roleMap["org_id"].(string)),
+						GroupId: conversion.StringPtr(roleMap["group_id"].(string)),
+						Role:    conversion.StringPtr(role.(string)),
+					}
+					roleAssignments = append(roleAssignments, roleAssignment)
+				}
+			}
+		}
+	}
+
+	sort.Sort(mRoleAssignment(roleAssignments))
```

Comment:
knit: i wouldn't have these empty lines or in next func

Reply:
Removed some empty lines in f605df2